sdk/bpf/c/inc/sol/string.h

```diff
@@ -45,14 +81,27 @@ static void *sol_memset(void *b, int c, size_t len) {
 
 /**
  * Find length of string
+ * Checks 4 chars at a time for faster performance
  */
 static size_t sol_strlen(const char *s) {
-  size_t len = 0;
-  while (*s) {
-    len++;
-    s++;
+  const char *ref = s;
+
+  if (!s) {
+    return 0;
+    }
+
+  while (1)
```

Comment:
Do you have evidence that this results in faster performance?

Reply:
I ran tests, yeah. It's faster because it checks 4 chars per loop instead of 1 resulting in less jmp which results in better performance + less branch misprediction. I  believe this is also how it's implement in glibc (though they do some magic there that makes it too unstable for this) Here is the source code https://github.com/lattera/glibc/blob/master/string/strlen.c

Reply:
Not exactly sure hpw to make those tests pass